plugins/woo-ai/src/product-description/product-description-button-container.tsx

```diff
@@ -0,0 +1,119 @@
+/**
+ * External dependencies
+ */
+import apiFetch from '@wordpress/api-fetch';
+import { __ } from '@wordpress/i18n';
+import { useState, useEffect, useRef } from '@wordpress/element';
+
+/**
+ * Internal dependencies
+ */
+import { useTinyEditor } from '../hooks/useTinyEditor';
+import MagicIcon from '../../assets/images/icons/magic.svg';
+
+type WooApiMessage = {
+	role: string;
+	content: string;
+};
+
+type WooApiResponse = {
+	generated_text: string;
+	previous_messages: WooApiMessage[];
+};
+
+const generatingContentPhrases = [
+	__( 'Please wait…', 'woocommerce' ),
+	__( 'Just a little longer…', 'woocommerce' ),
+	__( 'Almost there…', 'woocommerce' ),
+];
+
+const getGeneratingContentPhrase = () =>
+	generatingContentPhrases[
+		Math.floor( Math.random() * generatingContentPhrases.length )
+	];
+
+const DESCRIPTION_MAX_LENGTH = 300;
```

Comment:
What's the thought behind making this a constant vs having it embedded in the prompt directly? It doesn't make much difference but having it as a constant suggests that it's open to being changed at some point or used in some other capacity?

Reply:
Sounds good 👍